tasks/Task.cpp

```diff
@@ -26,16 +26,21 @@ Task::~Task()
 
 bool Task::configureHook()
 {
+    if (!_io_port.get().empty())
+      lamps.openURI(_io_port.value());
+    else
+      throw std::runtime_error("io_port property not set");
+
+    setDriver(&lamps);
+
     if (! TaskBase::configureHook())
         return false;
 
-    lamps.openURI(_port.get());
-
     led_list = _led_list.get();
     light_level_all  = _light_level_all.get();
 
     /* Sets initializing properties of the lamps */
-    for (int i = 0; i < led_list.size(); ++i)
+    for (unsigned int i = 0; i < led_list.size(); ++i)
     {
     	lamps.setPowerUpLightLevel(led_list[i].power_up_light_level, led_list[i].address);
```

Comment:
I've only edited this driver to include the iodrivers_base inheritance and haven't discussed its methods with anyone, but I guess it's only configuring the light level for once the led is activated (not actually turning it on).